### ISSUE TITLE
headless: -xml option returns Exception and in fact run an experiment

### ISSUE THREAD

**Issue by benoitgaudou** [gama-platform/gama#2824]
**Describe the bug**
I tried to use the option `-xml` of the headless, when GAMA returns an Exception and does not try to create a xml file representing the experiment, but try to run a simulation.

**To Reproduce**
1. On MacOs, GAMA (in its continuous built version) is "installed" in `/Applications` folder.
2. From a Terminal, I locate myself in the headless folder:
```
cd /Applications/GAMA1.8_Continuous_withJDK_Mac_07.28.19_23f1a7e.app/Contents/headless/
```
3. Now I have a look at the headless help related to the `-xml` option:
```
-xml	[experimentName] [modelFile.gaml] [xmlOutputFile.xml]
				        --  build an xml parameter file from a model
```
4. So I run the following instruction (note that I tried with `bash` and `sh` with the same result):
```
bash ./gama-headless.sh -xml prey_predatorExp samples/predatorPrey/predatorPrey.gaml toto.xml
```
5. I get in the terminal:
```
bash ./gama-headless.sh -xml prey_predatorExp samples/predatorPrey/predatorPrey.gaml toto.xml
./gama-headless.sh
******************************************************************
* GAMA version 1.8                                               *
* http://gama-platform.org                                       *
* (c) 2007-2019 UMI 209 UMMISCO IRD/SU & Partners                *
******************************************************************
GAMA is starting...
2019-07-29 10:43:43.299 java[17898:535442] [JRSAppKitAWT markAppIsDaemon]: Process manager already initialized: can't fully enable headless mode.
> JAI/ImageIO subsystem activated
Number of cpus used:8 (available: 8)
[Fatal Error] :1:1: Content is not allowed in prolog.
org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.
	at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:257)
	at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:339)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:121)
	at msi.gama.headless.xml.Reader.parseXmlFile(Reader.java:227)
	at msi.gama.headless.runtime.Application.runSimulation(Application.java:339)
	at msi.gama.headless.runtime.Application.start(Application.java:207)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:661)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:597)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1476)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1449)
	at org.eclipse.core.launcher.Main.main(Main.java:44)
An error has occurred. See the log file
/Applications/GAMA1.8_Continuous_withJDK_Mac_07.28.19_23f1a7e.app/Contents/headless/toto.xml/.work/.metadata/.log.
```

I also noticed that `toto.xml` is, in fact, a directory (and not a file), that contains the `snapshot` folder: this means that the headless is considering it as the folder in which it will put the result of a headless simulation.

**Expected behavior**
No exception and the `toto.xml` to be a file.

**Desktop (please complete the following information):**
 - OS: macOS
 - GAMA version: continuous built (GAMA1.8_Continuous_withJDK_Mac_07.28.19_23f1a7e)

**Comment by RoiArthurB** [comment]
Hi,

## TL;DR

You've encounter a known bug. We are currently working on it on another branch ! ^^

## Details about this bug

On the actual script, just some arguments are considered (`-m`, `-c`, `-help`, `-hpc`, `-p` and `-v`). Every other options are just dropped and not used. So the command you've run is considered like if you didn't used the `-xml` option :

```
./gama-headless.sh prey_predatorExp samples/predatorPrey/predatorPrey.gaml toto.xml
```
This line on the script will just keep 2 last arguments as input and output file, so : 
 - `samples/predatorPrey/predatorPrey.gaml` is your XML input file 
 - `toto.xml` is your output directory (on unix system you can create directory with extension without problem, so it's possible to have a directory named "toto.xml/")

Also, you should have a sub-directory (in _toto.xml/_) named ".work". The problem on the current script is that you indicate a location where the script will create the output directory (and it will never be a file), so it's not possible to generate XML file...

## Workaround

You can try the script from the _Release Candidate 2_ which looks more flexible. I haven't try it, maybe you'll have some bug with missing jar... Not sure.
Another way will be to update your current script with alpha script on **gama#gaml-headless** branch. @chapuisk and I are currently working on it, so it may contain some bugs, but you should be able to use `-xml` options :smiley:

**Comment by hqnghi88** [comment]
Hi, sorry not related with this issue but:
As we planned to try one more time to release 1.8 official this Wednesday, so did you two planned to finish your development in tuesday or to merge today your work into Master branch and continue on it ?

**Comment by RoiArthurB** [comment]
I don't know what you think @chapuisk, but personally I don't think that we'll finish what we're working on in the headless part on time... At least I can update the script for the release, but we won't have enough time to test everything and release a new headless 100% bug free :/ 
But let's talk about it by mail and leave this issue clean from this conversation ! ^^

@benoitgaudou does the RC2 script fixed your issue?

**Comment by chapuisk** [comment]
@hqnghi88 @benoitgaudou @RoiArthurB The dedicated [branch](https://github.com/gama-platform/gama/tree/gaml-headless) to fix and add features to headless is not meant to be in the upcoming release. Concerning -xml there is to bug fix: 1) suppress the option 2) update the current script to add -xml options and args to the $PARAM variable

**Comment by benoitgaudou** [comment]
The script of GAMA RC2 allows to run the `-xml` option with the command line I wrote previously.
But the script of the current Git version has definitely improved in terms of parameter checks and information provided to the user.

So it would be better to fix the current script :-)

**Comment by gnoubi** [comment]
Hi 

I intend to check the script today

Nicolas 

Nicolas Marilleau
UMI 209 UMMISCO IRD/UPMC
Chercheur invité Femto-st

Institut de Recherche pour le Développement 
32 avenue Henri Varagnat
93143 Bondy 
Fixe : 0148027901
Mobile: 0688334906

**Comment by benoitgaudou** [comment]
Fixed by several commits : the strategy has been to reuse the script used in the RC2 version instead of the new one.
